src/editor/autocomplete.js

```diff
@@ -42,29 +42,33 @@ export default class AutocompleteWrapperModel {
     async onTab(e) {
         const acComponent = this._getAutocompleterComponent();
 
-        if (acComponent.state.completionList.length === 0) {
+        if (acComponent.countCompletions() === 0) {
             // Force completions to show for the text currently entered
             await acComponent.forceComplete();
             // Select the first item by moving "down"
-            await acComponent.onDownArrow();
+            await acComponent.moveSelection(+1);
         } else {
-            if (e.shiftKey) {
-                await acComponent.onUpArrow();
-            } else {
-                await acComponent.onDownArrow();
-            }
+            await acComponent.moveSelection(e.shiftKey ? -1 : +1);
         }
         this._updateCallback({
             close: true,
         });
     }
 
     onUpArrow() {
-        this._getAutocompleterComponent().onUpArrow();
+        this._getAutocompleterComponent().moveSelection(-1);
     }
 
     onDownArrow() {
-        this._getAutocompleterComponent().onDownArrow();
+        this._getAutocompleterComponent().moveSelection(+1);
+    }
+
+    onLeftArrow() {
```

Comment:
No need to add left/right arrow keys for the new composer yet, I need to test if it doesn't interfere with caret navigation. Let's just do enough here so it isn't broken.